catroid/src/main/java/org/catrobat/catroid/io/asynctasks/CopyProjectToExternalStorageAsyncTask.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.io.asynctasks;
+
+import android.support.annotation.VisibleForTesting;
+import android.util.Pair;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.io.ZipArchiver;
+import org.catrobat.catroid.utils.StatusBarNotificationManager;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.catrobat.catroid.common.Constants.CATROBAT_EXTENSION;
+import static org.catrobat.catroid.common.FlavoredConstants.DEFAULT_ROOT_DIRECTORY;
+import static org.catrobat.catroid.common.FlavoredConstants.EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY;
+
+public class CopyProjectToExternalStorageAsyncTask extends android.os.AsyncTask<Pair<String, Integer>, Integer, Long> {
+	@Override
+	protected Long doInBackground(Pair<String, Integer>... programNames) {
+		int count = programNames.length;
+		long totalSize = 0;
+		for (int i = 0; i < count; i++) {
+			copyProjectToExternalStorage(programNames[i].first, programNames[i].second);
+		}
+		return totalSize;
+	}
+
+	@VisibleForTesting
+	public void copyProjectToExternalStorage(String projectName, int notificationID) {
+		final File projectFile = new File(DEFAULT_ROOT_DIRECTORY, projectName);
+		final File externalProjectZip = new File(EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY,
+				projectName + CATROBAT_EXTENSION);
+
+		if (!EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY.exists()) {
+			EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY.mkdirs();
```

Comment:
its probably a good idea to check if mkdirs was successful and return if that isnt the case.

Reply:
These lines are not necessary, because the zipArchiver will create this folder if it doesn't exist.
(I will delete this if)